src/Rules/PHPUnit/CoversHelper.php

```diff
@@ -83,16 +85,22 @@ public function processCovers(
 
 		if ($this->reflectionProvider->hasClass($className)) {
 			$class = $this->reflectionProvider->getClass($className);
+
 			if (isset($method) && $method !== '' && !$class->hasMethod($method)) {
 				$errors[] = RuleErrorBuilder::message(sprintf(
 					'@covers value %s references an invalid method.',
 					$covers
 				))->build();
 			}
 		} else {
+			if (!isset($method) && $this->reflectionProvider->hasFunction(new Name($covers, []), null)) {
+				return $errors;
+			}
+
 			$errors[] = RuleErrorBuilder::message(sprintf(
-				'@covers value %s references an invalid class.',
-				$covers
+				'@covers value %s references an invalid %s.',
+				$covers,
+				$isMethod ? 'method' : 'class or function'
```

Comment:
I don't see any test for "class or function", can you find it or add it?

Reply:
good spot, i've added that now